pecos/core/utils/mmap_util.hpp

```diff
@@ -0,0 +1,530 @@
+        T fget_one() {
+            check_get_();
+            return mmap_r_->fget_one<T>();
+        }
+
+        template<class T, class TT=T, details_::if_simple_serializable<TT> = true>
+        T* fget_multiple(const uint64_t n_elements) {
+            check_get_();
+            return mmap_r_->fget_multiple<T>(n_elements);
+        }
+
+        template<class T, class TT=T, details_::if_simple_serializable<TT> = true>
+        void fput_one(const T& src) {
+            check_put_();
+            mmap_w_->fput_one<T>(src);
+        }
+
+        template<class T, class TT=T, details_::if_simple_serializable<TT> = true>
+        void fput_multiple(const T* src, const uint64_t n_elements) {
+            check_put_();
+            mmap_w_->fput_multiple<T>(src, n_elements);
+        }
+
+    private:
+        enum class Mode_ {
+            UNINIT = 0,             // Uninitialized
+            READONLY = 1,           // "r"
+            WRITEONLY = 2,          // "w"
+        };
+
+        details_::MmapStoreSave_* mmap_w_ = nullptr; // Used for write mode
+        details_::MmapStoreLoad_* mmap_r_ = nullptr; // Used for read mode
+        Mode_ mode_ = Mode_::UNINIT; // Indicator for current opened mode
+
+        /* Check if opened as read mode for calling get functions */
+        void check_get_() {
+            if (!mmap_r_) {
+                throw std::runtime_error("Not opened for read mode, cannot call get.");
+            }
+        }
+
+        /* Check if opened as write mode for calling put functions */
+        void check_put_() {
+            if (!mmap_w_) {
+                throw std::runtime_error("Not opened for write mode, cannot call put.");
+            }
+        }
+}; // end class MmapStore
+
+
+/* Extended Vector class that can be used as a std::vector or as a memory-view from part of mmap file
+ * For std::vector case, it own the memory for data storage.
+ * For mmap view case, it does not own any memory, but serve as a view for a piece of memory owned by MmapStore.
+ * By default, it is initialized as empty std::vector that can be resized or loaded as mmap view.
+ * Once loaded as mmap view, it cannot go back to std::vector case unless clear() is called.
+ */
+template<class T, class TT = T, details_::if_simple_serializable<TT> = true>
+class MmapableVector {
+    public:
+        MmapableVector() {} // Dummy constructor for empty case
+        MmapableVector(uint64_t size, const T& value) { resize(size, value); } // Constructor for vector case
+
+        /* Functions to match std::vector interface */
+        uint64_t size() const { return size_; }
+
+        T* data() { return data_; }
+        const T* data() const { return data_; }
+
+        T& operator[](uint64_t idx) { return data_[idx]; }
+        const T& operator[](uint64_t idx) const { return data_[idx]; }
+
+        bool empty() const { return static_cast<bool> (size_ == 0); }
+
+        void resize(uint64_t size, const T& value) {
```

Comment:
Let's add a default value in the signature so it can handle both cases. 
void resize(uint64_t size, const T& value=T() )